samcli/commands/delete/delete_context.py

```diff
@@ -108,38 +108,29 @@ def init_clients(self):
         """
         Initialize all the clients being used by sam delete.
         """
-        if not self.region:
-            if not self.no_prompts:
-                session = boto3.Session()
-                region = session.region_name
-                self.region = region if region else "us-east-1"
-            else:
-                # TODO: as part of the guided and non-guided context separation, we need also to move the options
-                # validations to a validator similar to samcli/lib/cli_validation/image_repository_validation.py.
-                raise click.BadOptionUsage(
-                    option_name="--region",
-                    message="Missing option '--region', region is required to run the non guided delete command.",
-                )
-
-        if self.profile:
-            Context.get_current_context().profile = self.profile
-        if self.region:
-            Context.get_current_context().region = self.region
-
-        boto_config = get_boto_config_with_user_agent()
-
-        # Define cf_client based on the region as different regions can have same stack-names
-        cloudformation_client = boto3.client(
-            "cloudformation", region_name=self.region if self.region else None, config=boto_config
+        client_provider = get_boto_client_provider_with_config(
+            region=self.region if self.region else None, profile=self.profile if self.profile else None
```

Comment:
nit: assuming these fields are instantiated as `None` default, we may simplify this line (not sure if that has some side effects)
```suggestion
            region=self.region, profile=self.profile
```